lib/decoder.js

```diff
@@ -0,0 +1,79 @@
+import { promisify } from "node:util";
+import { gunzip } from "node:zlib";
+import log from "lambda-log";
+import { toEpochMilliseconds } from "./timestamp";
+
+const gunzipPromise = promisify(gunzip);
+
+/**
+ * Parse lambda event kinesis records, and filter/log invalid
+ */
+export async function decodeRecords(event) {
```

Comment:
This is a mix of what was in `lib/get-records.js` and the old shared `index.js` handler.  Now that there are 4 separate handlers, much DRY-er to locate this all here.

The tests for this file likewise came out of the old tests for those 2 files.